File: java-algo/src/main/java/com/leetcode/Problem225.java
package com.leetcode;

import java.util.LinkedList;
import java.util.Queue;

/**
 * 225. Implement Stack using Queues
 * https://leetcode.com/problems/implement-stack-using-queues/
 */
public class Problem225 {
    private static class MyStack {

        private final Queue<Integer> queue = new LinkedList<>();

        public MyStack() {
        }

        public void push(int x) {
            queue.offer(x);
        }

        public int pop() {
            int n = queue.size(), i = 0;
            while (i < n - 1) {
                queue.offer(queue.poll());
                i++;
            }
            return queue.poll();
        }

        public int top() {
            int ret = 0;
            int n = queue.size();
            for (int i = 0; i < n; i++) {
                ret = queue.poll();
                queue.offer(ret);
            }
            return ret;
        }

        public boolean empty() {
            return queue.isEmpty();
        }
    }
}

File: python-algo/leetcode/problem_225.py
"""225. Implement Stack using Queues
"""


class MyStack:

    def __init__(self):
        self.q = [[], []]
        self.i = 0

    def push(self, x: int) -> None:
        self.q[self.i].append(x)

    def pop(self) -> int:
        while len(self.q[self.i]) > 1:
            num = self.q[self.i].pop(0)
            self.q[1 - self.i].append(num)
        ret = self.q[self.i].pop()
        self.i = 1 - self.i
        return ret

    def top(self) -> int:
        return self.q[self.i][-1]

    def empty(self) -> bool:
        return len(self.q[self.i]) == 0

